src/components/common/Dialogs/CreatePaymentDialog/CreatePaymentDialog.tsx

```diff
@@ -0,0 +1,248 @@
+  //   colonyData?.processedColony?.isWhitelistActivated;
+
+  // const verifiedUsers = useMemo(() => {
+  //   return getVerifiedUsers(colony.whitelistedAddresses, subscribedUsers) || [];
+  // }, [subscribedUsers, colony]);
+
+  // const showWarningForAddress = (walletAddress) => {
+  //   if (!walletAddress) return false;
+  //   return isWhitelistActivated
+  //     ? !colonyData?.processedColony?.whitelistedAddresses.some(
+  //         (el) => el.toLowerCase() === walletAddress.toLowerCase(),
+  //       )
+  //     : false;
+  // };
+
+  const transform = useCallback(
+    () =>
+      pipe(
+        mapPayload((payload) => {
+          const {
+            amount,
+            tokenAddress,
+            domainId,
+            recipient: {
+              profile: { walletAddress },
+            },
+            annotation: annotationMessage,
+            motionDomainId,
+          } = payload;
+
+          const selectedToken = colonyTokens.find(
+            (token) => token?.token.tokenAddress === tokenAddress,
+          );
+          const decimals = getTokenDecimalsWithFallback(
+            selectedToken?.token.decimals,
+          );
+
+          // const amountWithFees = networkFeeInverse
+          //   ? calculateFee(amount, networkFeeInverse, decimals).totalToPay
+          //   : amount;
+
+          return {
+            colonyName,
+            colonyAddress,
+            recipientAddress: walletAddress,
+            domainId,
+            singlePayment: {
+              tokenAddress,
+              amount, // amountWithFees - @NOTE: The contract only sees this amount
+              decimals,
+            },
+            annotationMessage,
+            motionDomainId,
+          };
+        }),
+        withMeta({ navigate }),
+      ),
+    [colonyAddress, colonyName, colonyTokens, navigate],
+  );
+
+  return (
+    <Form<FormValues>
+      defaultValues={{
+        forceAction: false,
+        domainId: (filteredDomainId === 0 || filteredDomainId === undefined
+          ? Id.RootDomain
+          : filteredDomainId
+        ).toString(),
+        recipient: undefined,
+        amount: 0,
+        tokenAddress: nativeToken.tokenAddress,
+        annotation: '',
+        motionDomainId:
+          filteredDomainId === 0 || filteredDomainId === undefined
+            ? Id.RootDomain
+            : filteredDomainId,
+      }}
+      validationSchema={validationSchema}
+      submit={getFormAction('SUBMIT')}
+      error={getFormAction('ERROR')}
+      success={getFormAction('SUCCESS')}
+      transform={transform}
+      onSuccess={close}
+    >
+      {({ getValues, formState, setValue }) => {
+        const values = getValues();
+        if (values.forceAction !== isForce) {
+          setIsForce(values.forceAction);
+        }
+
+        return (
+          <Dialog cancel={cancel}>
+            <DialogForm
+              {...formState}
+              values={values}
+              setValue={setValue}
```

Comment:
Is our aim to just not pass values as props that can be obtained through hooks? Jakub mentioned that we are moving out from passing `colony` as a prop, and now this. 

I'm asking this because I'm wondering if that's really worth it from a performance point of view as we would suddenly have lots of hooks requesting values that may already be available 1 level above

Reply:
I think it depends on the situation. Here I agree with Will that it makes sense to call the hook within the consuming component rather than pass the props.
What concerns about performance do you have specifically? Will using hooks in multiple places impact it negatively? I wouldn't think it'd make a huge difference, at the end of the day whenever the value of the context changes, `DialogForm` will be re-rendered, no matter if it's getting the values via props or a hook

Reply:
Yeah, my concern was mostly thinking if it was worth it using a hook to get data that is being used anyways by the parent component. 

I did a little search and read some stuff about prop drilling vs using context, and looks like in our case performance wouldn't be a significant issue on whether we use one or the other. In general, so long as the data provided by the context is not something that is constantly changing, we are good.

That, plus the other benefits of using context in comparison with prop drilling, I would have to agree with you guys on using contexts more :)